WordPress/Sniffs/WP/ClassNameCaseSniff.php

```diff
@@ -0,0 +1,416 @@
+<?php
+/**
+ * WordPress Coding Standard.
+ *
+ * @package WPCS\WordPressCodingStandards
+ * @link    https://github.com/WordPress/WordPress-Coding-Standards
+ * @license https://opensource.org/licenses/MIT MIT
+ */
+
+namespace WordPressCS\WordPress\Sniffs\WP;
+
+use WordPressCS\WordPress\AbstractClassRestrictionsSniff;
+
+/**
+ * Verify whether references to WP native classes use the proper casing for the class name.
+ *
+ * @package WPCS\WordPressCodingStandards
+ *
+ * @since 3.0.0
+ */
+class ClassNameCaseSniff extends AbstractClassRestrictionsSniff {
+
+	/**
+	 * List of all WP native classes.
+	 *
+	 * To be updated after every major release.
```

Comment:
We could use `@internal` tag here, like in the capabilities sniff, with the last updated WP version. 

Something like:

```php
	 * {@internal To be updated after every major release. Last updated for WordPress 6.1.0.}	
```